starlark/src/values/mod.rs

```diff
@@ -820,13 +839,21 @@ impl Value {
         stop: Option<Value>,
         stride: Option<Value>,
     ) -> ValueResult {
-        let borrowed = self.0.borrow_mut();
+        let borrowed = self.0.borrow();
         borrowed.slice(start, stop, stride)
     }
-    pub fn iter<'a>(&'a self) -> Result<Box<dyn Iterator<Item = Value> + 'a>, ValueError> {
-        let borrowed = self.0.borrow();
-        let v: Vec<Value> = borrowed.iter()?.collect();
-        Ok(Box::new(v.into_iter()))
+    pub fn iter<'a>(&'a self) -> Result<RefIterable<'a>, ValueError> {
```

Comment:
Sorry for taking so long to review that PR. It kind of disturb me a bit. I agree with you than collecting here is not really good. I need to convince myself that this is the best way we can do.

Reply:
Ok I am still not convinced this is the best we can do but since I cannot find a better way, I say we better go forward with your change.